### PR TITLE
Add fin as a submodule

### PR DESCRIPTION
**Pull request by JehandadKhan**
This PR removes the old fin code which was added verbatim and replaces it with a git submodule. This allows better tracking with the upstream repo which is developed independently.

### CODE REVIEW

**Comment by JehandadKhan** [comment]
In draft state until I verify behavior of Jenkins with git submodules.

**Comment by pfultz2** [comment]
Why is fin developed independently? If it needs to be then why doesn't MIOpen use fin with `find_package`?

Or is fin more tightly coupled with MIOpen that it needs to build with MIOpen? If so, then why is this in a separate repo?

**Comment by junliume** [comment]
@JehandadKhan "Tuna Fin Build Test" stage has failed, and I guess that's expected?

**Comment by cderb** [comment]
Completed work with converting Fin to a submodule.

**Comment by JehandadKhan** [comment]
@cderb @junliume I am unable to approve this PR, since I am the original author.

**Comment by cderb** [comment]
> @cderb @junliume I am unable to approve this PR, since I am the original author.

Curiously enough I can approve it even though I was the last to make changes.

**Comment by cderb** [comment]
@junliume This removes the Fin code copy from MIOpen, and instead it is tracked as a submodule. This change would remove the diffs we see in MIOpen when we change Fin. MIOpen changes here are for how the Jenkinsfile interacts with and builds the Fin repo.

File: Jenkinsfile [hunk 1]
```diff
@@ -130,9 +186,8 @@ def getDockerImage(Map conf=[:])
     def miotensile_version = conf.get("miotensile_version", "default") // deprecated
     def target_id = conf.get("target_id", "OFF") // deprecated
     def mlir_build = conf.get("mlir_build", "ON") // always ON
-    def build_fin = conf.get("build_fin", "OFF") // forcing this to be always on means fewer docker containers since this ensures all dependencies are present in the docker image
     def no_cache = conf.get("no_cache", false)
-    def dockerArgs = "--build-arg BUILDKIT_INLINE_CACHE=1 --build-arg PREFIX=${prefixpath} --build-arg GPU_ARCH='${gpu_arch}' --build-arg MIOTENSILE_VER='${miotensile_version}' --build-arg USE_TARGETID='${target_id}' --build-arg USE_MLIR='${mlir_build}' --build-arg USE_FIN='${build_fin}' "
+    def dockerArgs = "--build-arg BUILDKIT_INLINE_CACHE=1 --build-arg PREFIX=${prefixpath} --build-arg GPU_ARCH='${gpu_arch}' --build-arg MIOTENSILE_VER='${miotensile_version}' --build-arg USE_TARGETID='${target_id}' --build-arg USE_MLIR='${mlir_build}' "
```
Comment (atamazov) [inline Jenkinsfile:190]: Remove USE_FIN from Dockerfile.

Comment (atamazov) [reply Jenkinsfile:190]: @cderb Let's remove unused code.

Comment (atamazov) [reply Jenkinsfile:190]: @JehandadKhan oops

Comment (junliume) [reply Jenkinsfile:190]: @atamazov should `USE_FIN` be removed from everywhere in the repo? I may need to update dockerfile too.

Comment (atamazov) [reply Jenkinsfile:190]: @junliume I do not know. But removal of `USE_FIN='${build_fin}'` at this line in Jenkinsfile makes USE_FIN unused in the Dockerfile, so it should be removed.

**Pull request merged**